### PR TITLE
Tursi FuseSoC/Icarus-based Pickling Infrastructure

### PR DESCRIPTION
**Pull request by getziadz**
Changes for "pickling" infrastructure. Used to create https://github.com/PrincetonUniversity/OPDB

### CODE REVIEW

**Comment by Jbalkind** [comment]
Georgios could you give tursi a go from this branch to check I didn't break it when I rebased? We also need to make sure we that have copyright headers on anything you added that wasn't a core (I think that should just be a few python files though). If you add commits to your branch, they'll automatically be included in this PR.

**Comment by getziadz** [comment]
A few commits were missing. I have added the changes; which included copyright headers in some python files.

File: piton/design/include/piton_include.core [hunk 1]
```diff
@@ -5,10 +5,11 @@ description: Top-level Piton includes
...
     rtl:
         files:
-            - piton_system.vh: {is_include_file: true}
+            - iop.h: {is_includ_file: true}
             - network_define.v: {is_includ_file: true}
+            - piton_system.vh: {is_include_file: true}
             - sys.h: {is_includ_file: true}
-            - iop.h: {is_includ_file: true}
+            - tlu.h: {is_includ_file: true}
         depend:
```
Comment (olofk) [inline piton/design/include/piton_include.core:15]: Typos here. Missing an `e` in `include`. Couldn't figure out if this was addressed in a later commit

Comment (getziadz) [reply piton/design/include/piton_include.core:15]: Thanks Olof, I just pushed the commit ( c0aea52a6de05bdd86db2e673e94c57642b81a25 ) that fixes the typos.

**Comment by Jbalkind** [comment]
I think we're missing `rf_l15_lrsc_flag.v.pyv` in `piton/design/chip/tile/l15/rtl/l15.core` if we want to be compatible with the most recent release.

**Comment by Jbalkind** [comment]
Another thing I just realised: The preprocessing script is still in `piton/design/include/preproc.py` - we should probably move it to piton/tools/bin with the other python scripts

**Comment by Jbalkind** [comment]
Ok I believe all previous points have been addressed. There's a chance of some small regressions due to the includes and so on but I think it's all covered.

**Pull request merged**